src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -564,18 +564,32 @@ internal void ExecutePackageDownload(string name, PackageVersion version, string
                     }
                 }
 
-                var localPkgs = pmExt.PackageLoader.LocalPackages;
+                var pkgVersion = VersionUtilities.PartialParse(package.version);
 
+                var localPkgs = pmExt.PackageLoader.LocalPackages;
+                    
                 var uninstallsRequiringRestart = new List<Package>();
                 var uninstallRequiringUserModifications = new List<Package>();
                 var immediateUninstalls = new List<Package>();
+                var stdLibPackages = new List<Package>();
 
                 // if a package is already installed we need to uninstall it, allowing
                 // the user to cancel if they do not want to uninstall the package
-                foreach (var localPkg in version.full_dependency_ids.Select(dep => localPkgs.FirstOrDefault(v => v.ID == dep._id)))
+                var duplicateLoacalPackages = package.full_dependency_ids.Select(dep => localPkgs.FirstOrDefault(v => v.Name == dep.name));
+                foreach (var localPkg in duplicateLoacalPackages)
                 {
                     if (localPkg == null) continue;
 
+                    // should we also uninstall deps with the same version?
+                    //if (VersionUtilities.PartialParse(localPkg.VersionName) == pkgVersion) continue;
+
+                    if (Configuration.DebugModes.IsEnabled("PackageManagerVsStdLibDebugMode") &&
+                        localPkg.RootDirectory.Contains(pmExt.PackageLoader.StandardLibraryDirectory))
```

Comment:
we should probably make this a helper method, seems we're going to use it all over.